src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs

```diff
@@ -69,8 +86,23 @@ private void ReportDiagnosticsAndEmitSource(SourceProductionContext sourceProduc
             }
 
             OnSourceEmitting?.Invoke(ImmutableArray.Create(input.ContextGenerationSpec));
-            Emitter emitter = new(sourceProductionContext);
-            emitter.Emit(input.ContextGenerationSpec);
+
+            // Ensure the source generator emits number literals using invariant culture.
+            // This prevents issues such as locale-specific negative signs (e.g., U+2212 in fi-FI)
+            // from being written to generated source files.
+#pragma warning disable RS1035 // CultureInfo.CurrentCulture is banned in analyzers
```

Comment:
I would much rather this involved a targeted fix of the issue at hand, i.e. using `SymbolDisplay.FormatPrimitive` instead of `ToString()` wherever numeric literals are being formatted.